### ISSUE TITLE
Ignore directories in modpacks that begin with . or _

### ISSUE THREAD

**Issue by sbrl** [minetest/minetest#11240]
## Problem
When developing a mod that is installed via git (e.g. on a server), it is desirable to have subdirectories for non-mod things, such as documentation (e.g. a `docs` subdirectory), or tests (e.g. `tests`).

Unfortunately, Minetest tries to load mods from *every* subdirectory in a modpack, which prevents one from keeping documentation etc in it's own dedicated subdirectory.

## Solutions
The simplest solution here I can think of is to ignore all subdirectories that begin with a `.` or an `_` when loading the mods in a modpack.

This would allow one to store documentation in `_docs` and tests in `_tests` for example, thereby eliminating clutter.

## Alternatives
Another potential solution I thought of was to put the modpack in an `src` subdirectory. Unfortunately, this does not work if one clones the git repository into their `worldmod` directory, as Minetest looks in the top-level directory and not `src/`.

Additionally while ContentDB supports a `.gitattributes` file for excluding things for export, this is not the case when cloning a repository via git - which is commonly used both in development and on servers, so this does not solve the problem.

## Additional context
Specifically, I want to expand the documentation for my [WorldEditAdditions](https://github.com/sbrl/Minetest-WorldEditAdditions/) mod and put it in it's own subdirectory - perhaps using GitHub pages and a [static site generator](https://staticgen.com/) to make a website that's more user-friendly than a markdown document on GitHub.

**Comment by SmallJoker** [comment]
https://github.com/minetest/minetest/blob/de85bc9227ef0db01854fa0eef89256646b4d578/src/content/mods.cpp#L173-L177

This already happens for hidden (`.`) directories.

**Comment by Emojigit** [comment]
underscores are legal, so I changed the prefix to a tilde.

**Comment by sbrl** [comment]
Oh, I was not aware of dots `.` @SmallJoker! Thanks for that. Though things with dots at the beginning are hidden on Linux, which isn't desirable for documentation for example.

As mentioned by @ ryvnf [here](https://github.com/minetest/minetest/pull/11242#issuecomment-841706287), transparently ignoring directories that don't contain an `init.lua` might be the best solution to the issue.

**Comment by Emojigit** [comment]
> Transparently ignoring directories that don't contain an `init.lua` might be the best solution to the issue.

No, because then you may not find file naming issues (i.e. sometimes I name `init.lua` to `main.py`, etc XD)

**Comment by sbrl** [comment]
Oh, I see. Perhaps a comma-separated list of directories to ignore in `modpack.conf` then maybe?

**Comment by neoh4x0r** [comment]
@sbrl 
> Additionally while ContentDB supports a .gitattributes file for excluding things for export, this is not the case when cloning a 
> repository via git - which is commonly used both in development and on servers, so this does not solve the problem.

Git does support **.gitattributes** for checkout.
https://git-scm.com/docs/gitattributes

**These attributes affect how the contents stored in the repository are copied to the working tree**

You should also look at this **stackoverflow** post:
https://stackoverflow.com/questions/5494576/git-how-to-ignore-specify-files-for-checkout

Apparently **git archive** uses **.gitattributes** and **export-ignore** to exclude files.
the archive command would create a tar.gz file, containing only the files needed for deployment.

**Comment by yogwoggf** [comment]
Any updates on this? I really need to be able to ignore specific directories when my mod is loaded, and my workarounds are failing. I'd be willing to PR this.

**Comment by yogwoggf** [comment]
For my `node_modules` usecase, adding an empty `init.lua` to it makes Minetest load fine. Node also doesn't care about the empty `init.lua`, so this is a reasonable workaround. Will still like to implement something formal.

**Comment by wsor4035** [comment]
to post my thoughts from another platform. personally i think that doing an excludes(debate the name of the key, idc) line in modpack.conf as suggested in https://github.com/minetest/minetest/issues/11240#issuecomment-850005220 makes the most sense (without considering implementation details) since its outside of the directory.

examples
`excludes = modname` or
`excludes = modname1, modname2`

looking at core devs thoughts from the dead pr linked to this ( https://github.com/minetest/minetest/pull/11242 ):

ruben: https://github.com/minetest/minetest/pull/11242#issuecomment-955780687
as seen above, forcing a naming convention wouldnt work for everything. additionally, to me at least, it doesnt make sense to have another custom file .filename when minetest already has config files such as mod.conf and modpack.conf

sfan5: https://github.com/minetest/minetest/pull/11242#issuecomment-955780687
you may not want/or cannot have any files in a directory for some other program (this also sorta applies to rubens as well, but most software ignores dot files/etc)
